conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -349,17 +367,36 @@ private CodeBlock typeAwareAssignment(EnrichedField enriched, Type type, boolean
                     ConjureCollections.class,
                     spec.name,
                     Expressions.requireNonNull(spec.name, enriched.fieldName().get() + " cannot be null"));
-            return shouldClearFirst
-                    ? CodeBlocks.of(CodeBlocks.statement("this.$1N.clear()", spec.name), addStatement)
-                    : addStatement;
+            if (shouldClearFirst) {
+                return CodeBlock.builder()
+                        .beginControlFlow("if ($1N instanceof $2T)", spec.name, Collection.class)
```

Comment:
I think we’re trying to keep the dependency footprint small so don’t want to necessarily pull in Guava, but we could simplify this with something like `Lists.newArrayList(Iterable)` / `Sets.newHashSet(Iterable)`

Reply:
Added 2 methods to `ConjureCollections` to handle these and simplify generated code